src/main/java/edu/oakland/production/display01/DisplayGpsReceiverImplementation.java

```diff
@@ -1,20 +1,35 @@
 package edu.oakland.production.display01;
 
 import edu.oakland.helper.admin.Satellite;
+import edu.oakland.production.display01.DisplayGpsManager;
 import edu.oakland.helper.display01.SatelliteSignalCheckRequest;
+import java.lang.IllegalArgumentException;
 
 public class DisplayGpsReceiverImplementation implements DisplayGpsReceiver {
 
-  /** 
+  private DisplayGpsManager displayGpsManager;
+
+  /**
+   *
+   * @param displayGpsManager Calls on displayGpsManager to check value for null.
+   */
+  public DisplayGpsReceiverImplementation(DisplayGpsManager displayGpsManager) {
+    if (displayGpsManager == null) {
+      throw new IllegalArgumentException("Signal strength cannot be null");
+    }
+    this.displayGpsManager = displayGpsManager;
+  }
+
+  /**
    * Measures GPS Signal Strength.
    *
    * @param satelliteSignal GPS signal to measure.
-   * @return Status of GPS signal measured.
    */
   public String measureGpsSignalStrength(Satellite satelliteSignal) {
-
-    return "";
-
+    if (satelliteSignal == null) {
+      throw new IllegalArgumentException("Signal strength cannot be null");
```

Comment:
Just to make this exception a little more descriptive, can we change the output to the name of the parameter? Thank you!
```suggestion
      throw new IllegalArgumentException("satelliteSignal cannot be null");
```